### PR TITLE
[CMake] Establish a dependency from standard library variants onto 'libSwiftScan'

### PR DESCRIPTION
**Pull request by artemcm**
- [Move libSwiftScan from 'tools' to 'lib/Tooling'](https://github.com/swiftlang/swift/commit/8ce8af4029dfc31ff262929a988e3295b01c5298). Due to https://github.com/swiftlang/swift/blob/1965f96aff453778cf9f11e6a6ef16c904af9acd/CMakeLists.txt#L1475-L1500 we currently configure `stdlib` before `tools`. To bypass that, begin separating libraries used by the compiler into a separate directory in `lib`: `Tooling`, just like Clang does, and ensure they get configured before the standard library build.

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci smoke test

File: stdlib/public/core/CMakeLists.txt [hunk 1]
```diff
@@ -403,18 +403,14 @@ if(BOOTSTRAPPING_MODE STREQUAL "BOOTSTRAPPING")
...
-
 add_swift_target_library(swiftCore
                   ${SWIFT_STDLIB_LIBRARY_BUILD_TYPES}
                   ${swiftCore_common_options}
                   ${compile_flags_for_final_build}
                   FILE_DEPENDS
                     ${swiftCore_common_dependencies}
-                  DEPENDS ${tooling_stdlib_deps}
+                  DEPENDS
+                    libSwiftScan
```
Comment (etcwilde) [inline stdlib/public/core/CMakeLists.txt:413]: The freestanding configurations will likely fail with this since they don't build the tools.

Comment (etcwilde) [reply stdlib/public/core/CMakeLists.txt:413]: (alternatively said, they should not build tools nor tool libraries. They should only build the runtimes and use a pre-built toolchain.)

**Comment by artemcm** [comment]
@swift-ci smoke test

**Comment by artemcm** [comment]
@swift-ci smoke test

**Pull request merged**